extensions/eol-runtime-scanner/tests/test_version_ordering.R

```diff
@@ -0,0 +1,69 @@
+source("../version_ordering.R")
+
+versions <- c(
+  "3.8.2",
+  "2.8.9",
+  NA,
+  "",
+  "three",
+  "4.0.1",
+  "3.11.3",
+  "3.11.2",
+  "3.11.3",
+  "a.b.c",
+  "3.12.1"
+)
+
+test_that("sanitize_versions replaces malformed versions wth NA", {
+  expect_equal(
+    sanitize_versions(versions),
+    c(
+      "3.8.2",
+      "2.8.9",
+      NA,
+      NA,
+      NA,
+      "4.0.1",
+      "3.11.3",
+      "3.11.2",
+      "3.11.3",
+      NA,
+      "3.12.1"
+    )
+  )
+})
+
+test_that("sort_unique_versions returns a sorted vector of unique versions", {
+  sanitized <- sanitize_versions(versions)
+  expect_equal(
+    sort_unique_versions(sanitized),
+    c("2.8.9", "3.8.2", "3.11.2", "3.11.3", "3.12.1", "4.0.1")
+  )
+
+  # The input vector must already be sanitized.
+  expect_error(sort_unique_versions(versions))
```

Comment:
Could you assert some portion of the error you get here (so that we know we are getting the one we expect)?